libraries/replace_between.rb

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright:: 2018 Sous Chefs
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Filter to replace lines between two matches
+module Line
+  class Filter
+    def replace_between(current, args)
+      # replace the lines matching a pattern found between lines matching two patterns
+      # current is an array of lines
+      # args[0] is a pattern. Replace lines after this pattern
```

Comment:
This is really, handy, I know it'll slow things down slightly, but is it worth naming the arguments so they're easier to read?

Reply:
I'm not sure what you mean, the first thing the code does is assign the args.

      start_pattern = verify_kind(args[0], Regexp)
      end_pattern = verify_kind(args[1], Regexp)
      insert_array = prepare_insert_lines(args[2])
      ends = verify_one_of(args[3], [nil, :exclude, :first, :include, :last]) || :exclude
      options(args[4], safe: [true, false])

I thought those were mostly clear?  Did you mean something else.